.github/workflows/test_docker_starter.yml

```diff
@@ -13,4 +13,4 @@ jobs:
     - uses: actions/checkout@v2
 
     - name: Test ./start-test-env
-      run: ./start-test-env --help
+      run: ./start-test-env doctor
```

Comment:
@tkilias you are right, as you mentioned I think the problem is at least two fold:

1. Stage0: bash script(s) 
2. Stage1: python code/scripts

There could be `an` issue with docker in each stage right?
Judging from the ticket, it looked to me, that the error was coming from python. So I was thinking, whatever component
triggered the execution of the reported python code, could have run the _**doctor / health - check**_ first, couldn't it?

I also was thinking, by separating the checks for python and shell-scripts we make sure, we do not have implicit dependencies about environment assumptions. Each `Stage` "knows" what to check, so it also will be simpler to replace
a component in the future (@tomuben mentioned that we are considering replacing the scripts at some point).

It still would be possible to call the `Stage1` checks within `Stage0` before execution is "handed over'.
e.g.:

```console
# make sure the shell scripts have what they need
check_pre_conditions_of_shell_scripts()

# arbitrary other code
...

# check the health of the "python execution environment"
./start-test-env doctor
```

Reply:
> maybe use another name
> 
> health, diag or so

I would go with `health` then

Reply:
@tkilias would you only rename the command name or also the "moudle" etc?

Reply:
Regarding the checks, yes we have two stages and you are right, each stage should check it independently, such that we don't make assumptions. We can't resuse the python checks for the bash scripts, because the assumption of the bash script is, that only docker and readlink are available. Otherwise, we wouldn't need the bash scripts at all. Their purpose is to reduce the required dependencies.

Reply:
Regarding the renaming, It is probably better to rename both, because otherwise don't find the corresponding things later.